### DISCUSSION TITLE
New security rules

### DISCUSSION THREAD

**Discussion by icy17** [openssl/openssl#25058] [Q&A]
Hi! I found that some[ security rules](https://github.com/openssl/openssl/discussions/24627) missing in the documentations. I'm tring to add these missing security rules. However, I'm not sure how to submit patches to supplement the documentation, so I'll provide some types of rules I've written based on other documentations. I'm not certain if they are appropriate, and if you find them acceptable, I can provide additional rules for other APIs in the same manner.
For API `BIO_free`: The BIO should be obtained from an API like BIO_new, and not previously freed. The caller should not attempt to access or dereference the BIO after invoking `BIO_free`.
For API `EVP_DigestInit`: The first parameter must not be NULL. Alternatively, this note can be added to the API `EVP_MD_CTX_new`: The caller must check the return value against NULL.

Additionally, I noticed that the example for the API `EVP_DigestInit_ex2` is missing a NULL check:
```
mdctx = EVP_MD_CTX_new();
    if (!EVP_DigestInit_ex2(mdctx, md, NULL)) {
        printf("Message digest initialization failed.\n");
        EVP_MD_CTX_free(mdctx);
        exit(1);
    }
```

**Comment by mattcaswell** [comment]
In order to make contributions please read the CONTRIBUTING file:

https://github.com/openssl/openssl/blob/master/CONTRIBUTING.md

All OpenSSL documentation is in the doc sub-directory of the main source distribution in a series of .pod files. For example the EVP_DigestInit_ex2 documentation is in this pod file:

https://github.com/openssl/openssl/blob/master/doc/man3/EVP_DigestInit.pod?plain=1

To contribute documentation update suggestions please make your changes to the relevant pod files and create a github pull request so that they can be reviewed.

**Comment by t8m** [comment]
I think that documenting which arguments must not be NULL would be useful addition to the respective manual pages for these functions. However I am not sure explicitly saying that you must not use objects after freeing them in every manual page of _free() function is really worth it. It would just add obvious things to the documentation.

**Comment by icy17** [comment]
Thank you for your suggestions. I will submit a pull request.
